paramparse: ensure `ttl = 'off'` works correctly

otan opened the pull request:
This was a missed item when implementing RESET (ttl) - `SET (ttl =
'off')` should work as well.

Release justification: high pri bug fix for new functionality

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/77650)
<!-- Reviewable:end -->

rafiss commented on pkg/sql/logictest/testdata/logic_test/row_level_ttl:
```diff
@@ -121,6 +120,34 @@ WHERE label LIKE 'row-level-ttl-%'
+# Check the same thing with SET (ttl = off)
+statement ok
+DROP TABLE tbl;
+CREATE TABLE tbl (
```
if you make DROP/CREATE two separate statements then you'll save me another edit on https://github.com/cockroachdb/cockroach/pull/76834 :) :) :)

otan replied:
you got it!

rafiss reviewed (approved):
lgtm!

otan committed:
paramparse: ensure `ttl = 'off'` works correctly

This was a missed item when implementing RESET (ttl) - `SET (ttl =
'off')` should work as well.

Release justification: high pri bug fix for new functionality

Release note: None

otan commented:
bors r=rafiss
